lib/improver/tests/cube_combiner/test_CubeCombiner.py

```diff
@@ -76,242 +72,134 @@ def test_basic(self):
         operation = '*'
         plugin = CubeCombiner(operation)
-        cube1 = self.cube1
-        cube2 = cube1.copy()
-        result = plugin.combine(cube1, cube2, operation)
+        result = plugin.combine(self.cube1, self.cube2)
         self.assertIsInstance(result, Cube)
-        expected_data = np.zeros((1, 2, 2, 2))
-        expected_data[0, 0, :, :] = 0.25
-        expected_data[0, 1, :, :] = 0.36
+        expected_data = np.full((1, 2, 2), 0.3, dtype=np.float32)
         self.assertArrayAlmostEqual(result.data, expected_data)
 
     def test_add(self):
         """Test combine adds the cubes correctly. """
         operation = '+'
         plugin = CubeCombiner(operation)
-        result = plugin.combine(self.cube1, self.cube2,
-                                operation)
-        expected_data = np.zeros((1, 2, 2, 2))
-        expected_data[0, 0, :, :] = 0.6
-        expected_data[0, 1, :, :] = 1.0
+        result = plugin.combine(self.cube1, self.cube2)
+        expected_data = np.full((1, 2, 2), 1.1, dtype=np.float32)
         self.assertArrayAlmostEqual(result.data, expected_data)
 
     def test_minus(self):
         """Test combine minus the cubes correctly. """
         operation = '-'
         plugin = CubeCombiner(operation)
-        result = plugin.combine(self.cube1, self.cube2,
-                                operation)
-        expected_data = np.zeros((1, 2, 2, 2))
-        expected_data[0, 0, :, :] = 0.4
-        expected_data[0, 1, :, :] = 0.2
+        result = plugin.combine(self.cube1, self.cube2)
+        expected_data = np.full((1, 2, 2), -0.1, dtype=np.float32)
         self.assertArrayAlmostEqual(result.data, expected_data)
 
     def test_max(self):
         """Test combine finds the max of the cubes correctly."""
         operation = 'max'
         plugin = CubeCombiner(operation)
-        result = plugin.combine(self.cube1, self.cube3,
-                                operation)
-        expected_data = np.zeros((1, 2, 2, 2))
-        expected_data[0, 0, :, :] = 0.5
-        expected_data[0, 1, :, :] = 0.8
+        result = plugin.combine(self.cube1, self.cube2)
+        expected_data = np.full((1, 2, 2), 0.6, dtype=np.float32)
         self.assertArrayAlmostEqual(result.data, expected_data)
 
     def test_min(self):
         """Test combine finds the min of the cubes correctly."""
         operation = 'min'
         plugin = CubeCombiner(operation)
-        result = plugin.combine(self.cube1, self.cube3,
-                                operation)
-        expected_data = np.zeros((1, 2, 2, 2))
-        expected_data[0, 0, :, :] = 0.1
-        expected_data[0, 1, :, :] = 0.6
+        result = plugin.combine(self.cube1, self.cube2)
+        expected_data = np.full((1, 2, 2), 0.5, dtype=np.float32)
         self.assertArrayAlmostEqual(result.data, expected_data)
 
     def test_mean(self):
         """Test that the function adds the cubes correctly for mean."""
         operation = '+'
```

Comment:
Should this actually be taking in "mean" as an operation?

Reply:
Nope. This has been written such that the combine step does the addition for calculating means and then the division occurs in the CubeCombiner process method. As such, for this little function mean is simply an alias for addition.

Reply:
And now I've actually fixed it.